### PR TITLE
Use Candlepin dedicated repository

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: packages/katello/katello-repos/katello-repos.spec [hunk 1]
```diff
@@ -1,11 +1,12 @@
 %global pulpcore_version 3.39
+%global candlepin_version 4.3
```
Comment (ehelms) [inline packages/katello/katello-repos/katello-repos.spec:2]: I am not sure if I should start with 4.3 here or lean into the nightly version of the repository for `rpm/develop`.

File: packages/katello/katello-repos/katello.repo [hunk 2]
```diff
@@ -10,10 +10,10 @@ gpgcheck=@REPO_GPGCHECK@
 # Candlepin RPMs as supported by Katello - This is a coordinated
 # copy of Candlepin's packages in order to ensure compatibility
 
-[katello-candlepin]
+[candlepin]
 name=Candlepin: an open source entitlement management system.
-baseurl=https://yum.theforeman.org/katello/@REPO_VERSION@/candlepin/@DIST@/$basearch/
-gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman
+baseurl=https://yum.theforeman.org/candlepin/@CANDLEPIN_VERSION@/@DIST@/$basearch/
+gpgkey=https://yum.theforeman.org/candlepin/@CANDLEPIN_VERSION@/GPG-RPM-KEY-candlepin
```
Comment (ehelms) [inline packages/katello/katello-repos/katello.repo:16]: Copying how Pulpcore is setup in this file right now. There remains the question of how this GPG key ends up here in our newer process with Copr.

I am thinking I may need to add a step that copies the public GPG key into the repository structure during `build_stage_repository` action.

Comment (ekohl) [reply packages/katello/katello-repos/katello.repo:16]: Including a file is more reliable. Dnf doesn't check expirations and doesn't refresh if the URL didn't change. I ran into that on Fedora with my personal copr

File: packages/katello/katello-repos/katello-repos.spec (hunk 1 above)
Comment (ekohl) [reply packages/katello/katello-repos/katello-repos.spec:2]: I'd be fine with 4.3 so we can pick it and then update to nightly in a separate bump. But no strong feelings here

**Comment by ehelms** [comment]
This is a related change that is needed -- https://github.com/theforeman/theforeman-rel-eng/pull/313

**Review by Odilhao — APPROVED** [review]

**Comment by Odilhao** [comment]
Just hit this while trying to release Katello

https://jenkins-foreman.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/foreman-pipeline-katello-rpm-4.11/detail/foreman-pipeline-katello-rpm-4.11/1/pipeline/93

**Pull request merged**